numerical_methods/matrix_transpose.c

```diff
@@ -0,0 +1,135 @@
+#include <assert.h>
```

Comment:
Please add a detailed description of the implementation, detailed description, Wikipedia link, author, etc. etc.. You can see other files to get an idea/example of how you can do it. 🙂